### PR TITLE
Blog: Trubar

### PR DESCRIPTION
**Pull request by janezd**
To partially relieve @zalagruden from @BlazZupan's wrath. :)

Draft. Comment, correct - or reject if you think it doesn't belong to Orange blog.

### CODE REVIEW

File: content/blog/2023/2023-01-17-trubar.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+            Select full rows: Izbiraj cele vrstice
+            Restore Original Order: Izvirni vrstni red
+        def `set_dataset`:
+            name: false
+            Data: Podatki
+```
+
+If it looks simple, it's because [it *is* simple](http://janezd.github.io/trubar/getting-started/).
+
+Besides working with f-strings, the advantage of Trubar's approach is that is does not pollute the sources with calls and parentheses. There's enough of them already. Instead of marking strings for translation within the source code, they are marked (for non-translation) in the message file. This must be done at some point, but it's better to do it in a dedicated place, not in the code.
```
Comment (zalagruden) [inline content/blog/2023/2023-01-17-trubar.md:67]: "The advantage of Trubar's approach is that is does not pollute the sources with calls and parentheses."
As gettext does?

File: content/blog/2023/2023-01-17-trubar.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+Gettext-like approaches thus can't work on f-strings. To our knowledge, there is currently no tool to support translation of f-strings.
+
+Luckily, we are programmers.
+
+## Enters Trubar
+
+We developed Trubar. Named after a colorful translator and publisher from Slovenian history, Trubar collects all strings from sources and places them into a yaml-like file. There, they can be translated - or marked as strings that must not be translated. Then, Trubar copies the sources, substituting strings with their translations, if provided.
```
[the remark this answers is not in the document]
Comment (janezd) [reply content/blog/2023/2023-01-17-trubar.md:35]: "Colorful language" would sound foul to me as well, but colorful persons were, for me, always just ... interesting.

But Noah sided with you, too, so I changed it into "Named after Primož Trubar (the author of the first Slovenian printed books, Catechismus and Abecedarium, and the first translator of part of the Bible in Slovenian language), Trubar collects all strings from sources and places them into a yaml-like file." It's a bit dense, but more informative.

The pun used to be about the black-and-white image that was supposed to be shown at the top of the page. But our blogs don't have an image at the top, so it doesn't make sense anyway.

File: content/blog/2023/2023-01-17-trubar.md (hunk 1 above)
Comment (janezd) [reply content/blog/2023/2023-01-17-trubar.md:67]: Yes. I now added "- unlike gettext -". This sentence used to be closer to description of gettext, but now too much other content came in between so a reminder is needed.

File: content/blog/2023/2023-01-17-trubar.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+Developed by native speaker of this beautifully complicated Slavic language, Trubar handles it with excellence. The beauty of f-strings is that they make implementation of plural forms much simpler because they allow using arbitrary functions. We simply inserted the appropriate function call and that's it. For instances, besides having four plural forms, the translation of proposition "with" before a number depends upon the number, and this is handled simply by using the function that returns the proper former for the given number. Which all happens without any help in the original source.
+
+More about localization in [documentation](http://janezd.github.io/trubar/localization/).
+
+## Ready to use?
+
+Sure, try it out. Trubar is [pip-installable](https://pypi.org/project/trubar/), [documentation](http://janezd.github.io/trubar/) is decent enough and although it is still early in development, we don't plan any huge changes. The least we can promise is that its compatibility-breaking changes will give you less headache that those by pyqtgraph. (Though ... this may not be much of a promise. :)
```
[the remark this answers is not in the document]
Comment (janezd) [reply content/blog/2023/2023-01-17-trubar.md:95]: I thought you got notices from the repo.

Orange is translated, as well as the geo add-on. And a bit of prototypes (Quick Select).

**Pull request merged**